lib/support/GallerySetupUtilities.class

```diff
@@ -362,12 +362,12 @@ class GallerySetupUtilities {
 	 * attack.  Either create a clean session, or if we detect that this session is clean,
 	 * sign it in a way that we'll recognize.
 	 */
-	$remoteId = GallerySession::getRemoteIdentifier();
+	$remoteId = (new GallerySession)->getRemoteIdentifier();
 	$cutoff = time() - G2_SUPPORT_MAX_SESSION_LIFETIME;
 	$configDir = GallerySetupUtilities::getConfigDir();
 	if (!empty($sessionId) && (
 		    !isset($_SESSION['_remoteId'])
-		    || GallerySession::compareIdentifiers($_SESSION['_remoteId'], $remoteId) == 0
```

Comment:
Here I think we need to make `compareIdentifiers()` a static method on line 1306 of `modules/core/classes/GallerySession.class`. We'll probably also need to change line 251 of that same file to call this method with the static syntax.

Reply:
lgtm, happy for you to make the change directly, rather than me bumble around.